tools/python/boutiques/schema/descriptor.schema.json

```diff
@@ -49,29 +49,33 @@
             },{
                 "oneOf": [{
                     "properties": {
-                        "type": { "enum": ["docker"] },
+                        "type": { "enum": ["docker", "singularity"] },
                         "image": {
                             "id": "http://github.com/boutiques/boutiques-schema/container/image",
                             "minLength": 1,
-                            "description": "Name of an image where the tool is installed and configured. Example: docker.io/neurodebian.",
+                            "description": "Name of an image where the tool is installed and configured. Example: bids/mriqc.",
                             "type": "string"
                         },
+                        "entrypoint": {
+                          "id": "http://github.com/boutiques/boutiques-schema/container/entrypoint",
+                          "description": "Flag indicating whether or not the container uses an entrypoint.",
+                          "type": "boolean"
+                        },
                         "index": {
                             "id": "http://github.com/boutiques/boutiques-schema/container/index",
                             "minLength": 1,
-                            "description": "Index where the image is available.",
-                            "default": "http://index.docker.io",
+                            "description": "Optional index where the image is available, if not the standard location. Example: docker.io",
                             "type": "string"
-                        },           
+                        },
                         "working-directory": {},
                         "container-hash": {}
                     },
-                    "required": ["type", "image"],
```

Comment:
how about making entrypoint optional? It would default to false, which would make the minimal descriptor less verbose. And not break backward compatibility.